services/explorer/src/api.rs

```diff
@@ -571,13 +584,21 @@ impl ExplorerApi {
             sender
                 .broadcast_transaction(verified)
                 .map(move |_| TransactionResponse { tx_hash })
-                .map_err(|e| ApiError::InternalError(e.into()))
+                .map_err(|e| {
+                    api::Error::new(api::HttpStatusCode::INTERNAL_SERVER_ERROR)
+                        .title("Failed to add transaction")
+                        .detail(e.to_string())
+                })
         };
 
         Box::new(
             verify_message(snapshot, query.tx_body)
                 .into_future()
-                .map_err(|e| ApiError::BadRequest(e.to_string()))
+                .map_err(|e| {
+                    api::Error::new(api::HttpStatusCode::BAD_REQUEST)
+                        .title("Failed to add transaction")
```

Comment:
Nit: Slightly more verbose `title` (e.g., "Failed to add transaction to memory pool") would be more helpful to clients.